activelearning.py

```diff
@@ -1,8 +1,349 @@
-# Get initial data set
-for it in range(n_iter):
-    # 1. Train proxy
-    # 2. Train GFN
-    # 3. Sample queries from GFN
-    # 4. Get scores for queries from oracle
-    # 5. Augment data set
-    pass
+from omegaconf import DictConfig, OmegaConf
+import hydra
+
+#--------------------------------
+'''
+AL Global pipeline
+'''
+#TODO : instantiate the config with hydra . Once we have the config object we can pass it to other functions
+class ActiveLearning:
+    def __init__(self, config):
+        #assume we have a config object that we can pass to all the components of the pipeline like in the previous code, eg "config_test.yaml"
+        self.config = OmegaConf.load(config)
+        
+        #setup function that creates the directories to save data, ...
+        self.setup()
+        #util class to handle the transition format  
+        self.transition = FormatHandler()
+        #util class to handle the statistics during training
+        self.statistics = Statistics()
+
+        #load the main components of the AL pipeline
+        self.oracle = Oracle(self.config, self.transition)
+        self.proxy = Proxy(self.config, self.transition)
+        self.acq = AcquisitionFunction(self.config, self.proxy)
+        self.env = Env(self.config, self.acq)
+        self.gflownet = GFlowNet(self.config, self.env)
+        self.querier = Querier(self.config, self.gflownet)
+    
+    def runPipeline(self):
+        #useful variables
+        self.iter = None
+        self.terminal = 0
```

Comment:
What is `self.terminal` for?

Reply:
I took that from the previous code. In the previous code at the last iteration of the pipeline, we don't generate data to score because this data is not going to be used by the proxy to train again since the pipeline is over. I agree with you that we can remove that since even at the last iteration it is interesting to have the last queries of the best final version of the gfn

Reply:
Yes, let's remove this bit for now in this new version.